filter out alloc-free pairs from trace plot

waysg opened the pull request:
Summary:
When dealing with a large memory trace, the resulting plot can be challenging to interpret and analyze.
This commit introduces a feature that enables filtering of allocations that have already been freed, providing a more focused view.
The remaining events in the plot often warrant closer examination, as they may be indicative of potential out-of-memory (OOM) issues.

zdevito reviewed (approved):
Looks good! Thanks for adding it

waysg committed:
filter out alloc-free pairs from trace plot (#165752)

Summary:

When dealing with a large memory trace, the resulting plot can be challenging to interpret and analyze.
This commit introduces a feature that enables filtering of allocations that have already been freed, providing a more focused view.
The remaining events in the plot often warrant closer examination, as they may be indicative of potential out-of-memory (OOM) issues.

Test Plan:
Trace file https://www.internalfb.com/manifold/explorer/mvai_gpu_traces/tree/traces/dynocli/mvai-training-online-881944936/0/mem_snapshot/rank-0.Oct_06_03_16_06.2735.pickle

commands:
python3 _memory_viz.py trace_plot -f rank-0.Oct_06_03_16_06.2735.pickle -o trace.html


{F1982799113}


We used this feature to investigate a OOM issue
https://docs.google.com/document/d/1uMmgvYB_E30ocHv8_fajJ8lkmk6IXPWXydBqPZHFi-w/edit?tab=t.0

Reviewed By: zdevito

Differential Revision: D84892600
